doc/concepts.rst

```diff
@@ -1299,6 +1300,39 @@ Foreign archives are particularly useful when embedding a library written in
 a foreign language and/or built with another build system. See
 :ref:`foreign-sandboxing` for more details.
 
+
+.. _foreign-objects:
+
+Foreign Objects
+---------------
+
+It's possible to specify native object files to be packaged with OCaml
+libraries or linked into OCaml executables. Do this by using the
+``foreign_objects`` field of the ``library`` or ``executable`` stanzas.
+For example:
+
+.. code:: scheme
+
+    (executable
+     (public_name main)
+     (foreign_objects foo bar))
+
+    (rule
+     (targets foo.o bar.o)
+     (deps foo.c bar.c)
+     (action (run ocamlopt %{deps})))
```

Comment:
can you use `%{cc}` here? using ocamlopt directly is bad practice (it's not available on bytecode only systems) so we don't want to encourage it.